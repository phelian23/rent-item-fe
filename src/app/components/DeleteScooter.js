import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { deleteScooterById, getAllScooters } from '../redux/actions/items';

const DeleteScooter = () => {
  const dispatch = useDispatch();
  const scooters = useSelector((state) => state.items.scooters);

  const navigate = useNavigate();

  useEffect(() => {
    dispatch(getAllScooters());
  }, [dispatch]);

  const handleDelete = (id) => {
    dispatch(deleteScooterById(id));
    setTimeout(() => {
      navigate('/');
    }, 3000);
  };

  return (
    <div>
      <div className="text-center p-3 header-det">
        <h1>Scooter List</h1>
      </div>
      <div className="list-wrapper">
        <ul className="d-flex list-con">
          {scooters.map((scooter) => (
            <li className="list-group-item col-md-4" key={scooter.id}>
              <div className="scoot-card">
                <img
                  className="card-img-top"
                  src={scooter.photo}
                  alt="Card cap"
                />
                <div className="card-body d-flex justify-content-between mt-4">
                  <h5 className="card-title">{scooter.name}</h5>
                  <p className="card-text">{scooter.price}</p>
                </div>
                <button
                  type="submit"
                  className="btn btn-danger"
                  onClick={() => handleDelete(scooter.id)}
                >
                  Delete
                </button>
              </div>
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default DeleteScooter;
