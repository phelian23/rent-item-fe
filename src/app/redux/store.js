import { configureStore } from '@reduxjs/toolkit';
import { combineReducers, compose, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import logger from 'redux-logger';
import itemReducer from './reducers/itemReducer';
import userReducer from './reducers/userReducer';
import favouriteReducer from './reducers/favouriteReducer';

const reducer = combineReducers({
  items: itemReducer,
  user: userReducer,
  favourites: favouriteReducer,

  // Add other reducers here
});

const composed = compose(applyMiddleware(thunkMiddleware, logger));

const store = configureStore({ reducer, composed });

export default store;
