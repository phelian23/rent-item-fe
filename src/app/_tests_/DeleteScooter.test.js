import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import DeleteScooter from '../components/DeleteScooter';

describe('DeleteScooter', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <DeleteScooter />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
