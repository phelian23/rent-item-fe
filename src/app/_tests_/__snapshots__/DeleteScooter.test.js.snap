// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteScooter renders correctly 1`] = `
<div>
  <div
    className="text-center p-3 header-det"
  >
    <h1>
      Scooter List
    </h1>
  </div>
  <div
    className="list-wrapper"
  >
    <ul
      className="d-flex list-con"
    />
  </div>
</div>
`;
